From 2f8227361a671f41b21fdd70ee40866c5f32ee86 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Fri, 27 May 2022 10:42:08 +0200
Subject: [PATCH 1/2] libgomp: use _aligned_free in gomp_aligned_free()if
 needed

gomp_aligned_alloc() currently uses the Windows aligned allocation API
_aligned_malloc() when available, but unlike other APIs on other platform
this one allocates memory that can't be freed with free(), but requires
the special _aligned_free() or things can crash.

Fix by using _aligned_free() in gomp_aligned_free() if gomp_aligned_alloc()
uses _aligned_malloc().
---
 libgomp/alloc.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libgomp/alloc.c b/libgomp/alloc.c
index a2a25befdf3..67569192130 100644
--- a/libgomp/alloc.c
+++ b/libgomp/alloc.c
@@ -105,7 +105,11 @@ void
 gomp_aligned_free (void *ptr)
 {
 #ifdef GOMP_HAVE_EFFICIENT_ALIGNED_ALLOC
+#ifdef HAVE__ALIGNED_MALLOC
+  _aligned_free (ptr);
+#else
   free (ptr);
+#endif
 #else
   if (ptr)
     free (((void **) ptr)[-1]);
-- 
2.36.1

